// @aws-sdk/client-opensearch/dist/es/commands/AcceptInboundConnectionCommand.js
import { __extends } from "tslib";
import { AcceptInboundConnectionRequest, AcceptInboundConnectionResponse } from "../models/models_0";
import { deserializeAws_restJson1AcceptInboundConnectionCommand, serializeAws_restJson1AcceptInboundConnectionCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
var AcceptInboundConnectionCommand = (function (_super) {
    __extends(AcceptInboundConnectionCommand, _super);
    function AcceptInboundConnectionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    AcceptInboundConnectionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "OpenSearchClient";
        var commandName = "AcceptInboundConnectionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: AcceptInboundConnectionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: AcceptInboundConnectionResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    AcceptInboundConnectionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1AcceptInboundConnectionCommand(input, context);
    };
    AcceptInboundConnectionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1AcceptInboundConnectionCommand(output, context);
    };
    return AcceptInboundConnectionCommand;
}($Command));
export { AcceptInboundConnectionCommand };
