// @aws-sdk/client-opensearch/dist/es/commands/DescribeDomainsCommand.js
import { __extends } from "tslib";
import { DescribeDomainsRequest, DescribeDomainsResponse } from "../models/models_0";
import { deserializeAws_restJson1DescribeDomainsCommand, serializeAws_restJson1DescribeDomainsCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
var DescribeDomainsCommand = (function (_super) {
    __extends(DescribeDomainsCommand, _super);
    function DescribeDomainsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeDomainsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "OpenSearchClient";
        var commandName = "DescribeDomainsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeDomainsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeDomainsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeDomainsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeDomainsCommand(input, context);
    };
    DescribeDomainsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeDomainsCommand(output, context);
    };
    return DescribeDomainsCommand;
}($Command));
export { DescribeDomainsCommand };
