// @aws-sdk/client-opensearch/dist/es/commands/UpgradeDomainCommand.js
import { __extends } from "tslib";
import { UpgradeDomainRequest, UpgradeDomainResponse } from "../models/models_0";
import { deserializeAws_restJson1UpgradeDomainCommand, serializeAws_restJson1UpgradeDomainCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
var UpgradeDomainCommand = (function (_super) {
    __extends(UpgradeDomainCommand, _super);
    function UpgradeDomainCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpgradeDomainCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "OpenSearchClient";
        var commandName = "UpgradeDomainCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpgradeDomainRequest.filterSensitiveLog,
            outputFilterSensitiveLog: UpgradeDomainResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpgradeDomainCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1UpgradeDomainCommand(input, context);
    };
    UpgradeDomainCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1UpgradeDomainCommand(output, context);
    };
    return UpgradeDomainCommand;
}($Command));
export { UpgradeDomainCommand };
