// @aws-sdk/client-opensearch/dist/es/models/models_0.js
import { __assign } from "tslib";
import { SENSITIVE_STRING } from "@aws-sdk/smithy-client";
export var AcceptInboundConnectionRequest;
(function (AcceptInboundConnectionRequest) {
    AcceptInboundConnectionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AcceptInboundConnectionRequest || (AcceptInboundConnectionRequest = {}));
export var InboundConnectionStatusCode;
(function (InboundConnectionStatusCode) {
    InboundConnectionStatusCode["ACTIVE"] = "ACTIVE";
    InboundConnectionStatusCode["APPROVED"] = "APPROVED";
    InboundConnectionStatusCode["DELETED"] = "DELETED";
    InboundConnectionStatusCode["DELETING"] = "DELETING";
    InboundConnectionStatusCode["PENDING_ACCEPTANCE"] = "PENDING_ACCEPTANCE";
    InboundConnectionStatusCode["PROVISIONING"] = "PROVISIONING";
    InboundConnectionStatusCode["REJECTED"] = "REJECTED";
    InboundConnectionStatusCode["REJECTING"] = "REJECTING";
})(InboundConnectionStatusCode || (InboundConnectionStatusCode = {}));
export var InboundConnectionStatus;
(function (InboundConnectionStatus) {
    InboundConnectionStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InboundConnectionStatus || (InboundConnectionStatus = {}));
export var AWSDomainInformation;
(function (AWSDomainInformation) {
    AWSDomainInformation.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AWSDomainInformation || (AWSDomainInformation = {}));
export var DomainInformationContainer;
(function (DomainInformationContainer) {
    DomainInformationContainer.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DomainInformationContainer || (DomainInformationContainer = {}));
export var InboundConnection;
(function (InboundConnection) {
    InboundConnection.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InboundConnection || (InboundConnection = {}));
export var AcceptInboundConnectionResponse;
(function (AcceptInboundConnectionResponse) {
    AcceptInboundConnectionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AcceptInboundConnectionResponse || (AcceptInboundConnectionResponse = {}));
export var DisabledOperationException;
(function (DisabledOperationException) {
    DisabledOperationException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DisabledOperationException || (DisabledOperationException = {}));
export var LimitExceededException;
(function (LimitExceededException) {
    LimitExceededException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LimitExceededException || (LimitExceededException = {}));
export var ResourceNotFoundException;
(function (ResourceNotFoundException) {
    ResourceNotFoundException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ResourceNotFoundException || (ResourceNotFoundException = {}));
export var AccessDeniedException;
(function (AccessDeniedException) {
    AccessDeniedException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AccessDeniedException || (AccessDeniedException = {}));
export var OptionState;
(function (OptionState) {
    OptionState["Active"] = "Active";
    OptionState["Processing"] = "Processing";
    OptionState["RequiresIndexDocuments"] = "RequiresIndexDocuments";
})(OptionState || (OptionState = {}));
export var OptionStatus;
(function (OptionStatus) {
    OptionStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(OptionStatus || (OptionStatus = {}));
export var AccessPoliciesStatus;
(function (AccessPoliciesStatus) {
    AccessPoliciesStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AccessPoliciesStatus || (AccessPoliciesStatus = {}));
export var AdditionalLimit;
(function (AdditionalLimit) {
    AdditionalLimit.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AdditionalLimit || (AdditionalLimit = {}));
export var Tag;
(function (Tag) {
    Tag.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Tag || (Tag = {}));
export var AddTagsRequest;
(function (AddTagsRequest) {
    AddTagsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AddTagsRequest || (AddTagsRequest = {}));
export var BaseException;
(function (BaseException) {
    BaseException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BaseException || (BaseException = {}));
export var InternalException;
(function (InternalException) {
    InternalException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InternalException || (InternalException = {}));
export var ValidationException;
(function (ValidationException) {
    ValidationException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ValidationException || (ValidationException = {}));
export var AdvancedOptionsStatus;
(function (AdvancedOptionsStatus) {
    AdvancedOptionsStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AdvancedOptionsStatus || (AdvancedOptionsStatus = {}));
export var SAMLIdp;
(function (SAMLIdp) {
    SAMLIdp.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SAMLIdp || (SAMLIdp = {}));
export var SAMLOptionsOutput;
(function (SAMLOptionsOutput) {
    SAMLOptionsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SAMLOptionsOutput || (SAMLOptionsOutput = {}));
export var AdvancedSecurityOptions;
(function (AdvancedSecurityOptions) {
    AdvancedSecurityOptions.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AdvancedSecurityOptions || (AdvancedSecurityOptions = {}));
export var MasterUserOptions;
(function (MasterUserOptions) {
    MasterUserOptions.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.MasterUserName && { MasterUserName: SENSITIVE_STRING })), (obj.MasterUserPassword && { MasterUserPassword: SENSITIVE_STRING }))); };
})(MasterUserOptions || (MasterUserOptions = {}));
export var SAMLOptionsInput;
(function (SAMLOptionsInput) {
    SAMLOptionsInput.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.MasterUserName && { MasterUserName: SENSITIVE_STRING }))); };
})(SAMLOptionsInput || (SAMLOptionsInput = {}));
export var AdvancedSecurityOptionsInput;
(function (AdvancedSecurityOptionsInput) {
    AdvancedSecurityOptionsInput.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.MasterUserOptions && { MasterUserOptions: MasterUserOptions.filterSensitiveLog(obj.MasterUserOptions) })), (obj.SAMLOptions && { SAMLOptions: SAMLOptionsInput.filterSensitiveLog(obj.SAMLOptions) }))); };
})(AdvancedSecurityOptionsInput || (AdvancedSecurityOptionsInput = {}));
export var AdvancedSecurityOptionsStatus;
(function (AdvancedSecurityOptionsStatus) {
    AdvancedSecurityOptionsStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AdvancedSecurityOptionsStatus || (AdvancedSecurityOptionsStatus = {}));
export var AssociatePackageRequest;
(function (AssociatePackageRequest) {
    AssociatePackageRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AssociatePackageRequest || (AssociatePackageRequest = {}));
export var DomainPackageStatus;
(function (DomainPackageStatus) {
    DomainPackageStatus["ACTIVE"] = "ACTIVE";
    DomainPackageStatus["ASSOCIATING"] = "ASSOCIATING";
    DomainPackageStatus["ASSOCIATION_FAILED"] = "ASSOCIATION_FAILED";
    DomainPackageStatus["DISSOCIATING"] = "DISSOCIATING";
    DomainPackageStatus["DISSOCIATION_FAILED"] = "DISSOCIATION_FAILED";
})(DomainPackageStatus || (DomainPackageStatus = {}));
export var ErrorDetails;
(function (ErrorDetails) {
    ErrorDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ErrorDetails || (ErrorDetails = {}));
export var PackageType;
(function (PackageType) {
    PackageType["TXT_DICTIONARY"] = "TXT-DICTIONARY";
})(PackageType || (PackageType = {}));
export var DomainPackageDetails;
(function (DomainPackageDetails) {
    DomainPackageDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DomainPackageDetails || (DomainPackageDetails = {}));
export var AssociatePackageResponse;
(function (AssociatePackageResponse) {
    AssociatePackageResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AssociatePackageResponse || (AssociatePackageResponse = {}));
export var ConflictException;
(function (ConflictException) {
    ConflictException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConflictException || (ConflictException = {}));
export var CancelServiceSoftwareUpdateRequest;
(function (CancelServiceSoftwareUpdateRequest) {
    CancelServiceSoftwareUpdateRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CancelServiceSoftwareUpdateRequest || (CancelServiceSoftwareUpdateRequest = {}));
export var DeploymentStatus;
(function (DeploymentStatus) {
    DeploymentStatus["COMPLETED"] = "COMPLETED";
    DeploymentStatus["ELIGIBLE"] = "ELIGIBLE";
    DeploymentStatus["IN_PROGRESS"] = "IN_PROGRESS";
    DeploymentStatus["NOT_ELIGIBLE"] = "NOT_ELIGIBLE";
    DeploymentStatus["PENDING_UPDATE"] = "PENDING_UPDATE";
})(DeploymentStatus || (DeploymentStatus = {}));
export var ServiceSoftwareOptions;
(function (ServiceSoftwareOptions) {
    ServiceSoftwareOptions.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ServiceSoftwareOptions || (ServiceSoftwareOptions = {}));
export var CancelServiceSoftwareUpdateResponse;
(function (CancelServiceSoftwareUpdateResponse) {
    CancelServiceSoftwareUpdateResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CancelServiceSoftwareUpdateResponse || (CancelServiceSoftwareUpdateResponse = {}));
export var AutoTuneDesiredState;
(function (AutoTuneDesiredState) {
    AutoTuneDesiredState["DISABLED"] = "DISABLED";
    AutoTuneDesiredState["ENABLED"] = "ENABLED";
})(AutoTuneDesiredState || (AutoTuneDesiredState = {}));
export var TimeUnit;
(function (TimeUnit) {
    TimeUnit["HOURS"] = "HOURS";
})(TimeUnit || (TimeUnit = {}));
export var Duration;
(function (Duration) {
    Duration.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Duration || (Duration = {}));
export var AutoTuneMaintenanceSchedule;
(function (AutoTuneMaintenanceSchedule) {
    AutoTuneMaintenanceSchedule.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AutoTuneMaintenanceSchedule || (AutoTuneMaintenanceSchedule = {}));
export var AutoTuneOptionsInput;
(function (AutoTuneOptionsInput) {
    AutoTuneOptionsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AutoTuneOptionsInput || (AutoTuneOptionsInput = {}));
export var ColdStorageOptions;
(function (ColdStorageOptions) {
    ColdStorageOptions.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ColdStorageOptions || (ColdStorageOptions = {}));
export var OpenSearchPartitionInstanceType;
(function (OpenSearchPartitionInstanceType) {
    OpenSearchPartitionInstanceType["c4_2xlarge_search"] = "c4.2xlarge.search";
    OpenSearchPartitionInstanceType["c4_4xlarge_search"] = "c4.4xlarge.search";
    OpenSearchPartitionInstanceType["c4_8xlarge_search"] = "c4.8xlarge.search";
    OpenSearchPartitionInstanceType["c4_large_search"] = "c4.large.search";
    OpenSearchPartitionInstanceType["c4_xlarge_search"] = "c4.xlarge.search";
    OpenSearchPartitionInstanceType["c5_18xlarge_search"] = "c5.18xlarge.search";
    OpenSearchPartitionInstanceType["c5_2xlarge_search"] = "c5.2xlarge.search";
    OpenSearchPartitionInstanceType["c5_4xlarge_search"] = "c5.4xlarge.search";
    OpenSearchPartitionInstanceType["c5_9xlarge_search"] = "c5.9xlarge.search";
    OpenSearchPartitionInstanceType["c5_large_search"] = "c5.large.search";
    OpenSearchPartitionInstanceType["c5_xlarge_search"] = "c5.xlarge.search";
    OpenSearchPartitionInstanceType["c6g_12xlarge_search"] = "c6g.12xlarge.search";
    OpenSearchPartitionInstanceType["c6g_2xlarge_search"] = "c6g.2xlarge.search";
    OpenSearchPartitionInstanceType["c6g_4xlarge_search"] = "c6g.4xlarge.search";
    OpenSearchPartitionInstanceType["c6g_8xlarge_search"] = "c6g.8xlarge.search";
    OpenSearchPartitionInstanceType["c6g_large_search"] = "c6g.large.search";
    OpenSearchPartitionInstanceType["c6g_xlarge_search"] = "c6g.xlarge.search";
    OpenSearchPartitionInstanceType["d2_2xlarge_search"] = "d2.2xlarge.search";
    OpenSearchPartitionInstanceType["d2_4xlarge_search"] = "d2.4xlarge.search";
    OpenSearchPartitionInstanceType["d2_8xlarge_search"] = "d2.8xlarge.search";
    OpenSearchPartitionInstanceType["d2_xlarge_search"] = "d2.xlarge.search";
    OpenSearchPartitionInstanceType["i2_2xlarge_search"] = "i2.2xlarge.search";
    OpenSearchPartitionInstanceType["i2_xlarge_search"] = "i2.xlarge.search";
    OpenSearchPartitionInstanceType["i3_16xlarge_search"] = "i3.16xlarge.search";
    OpenSearchPartitionInstanceType["i3_2xlarge_search"] = "i3.2xlarge.search";
    OpenSearchPartitionInstanceType["i3_4xlarge_search"] = "i3.4xlarge.search";
    OpenSearchPartitionInstanceType["i3_8xlarge_search"] = "i3.8xlarge.search";
    OpenSearchPartitionInstanceType["i3_large_search"] = "i3.large.search";
    OpenSearchPartitionInstanceType["i3_xlarge_search"] = "i3.xlarge.search";
    OpenSearchPartitionInstanceType["m3_2xlarge_search"] = "m3.2xlarge.search";
    OpenSearchPartitionInstanceType["m3_large_search"] = "m3.large.search";
    OpenSearchPartitionInstanceType["m3_medium_search"] = "m3.medium.search";
    OpenSearchPartitionInstanceType["m3_xlarge_search"] = "m3.xlarge.search";
    OpenSearchPartitionInstanceType["m4_10xlarge_search"] = "m4.10xlarge.search";
    OpenSearchPartitionInstanceType["m4_2xlarge_search"] = "m4.2xlarge.search";
    OpenSearchPartitionInstanceType["m4_4xlarge_search"] = "m4.4xlarge.search";
    OpenSearchPartitionInstanceType["m4_large_search"] = "m4.large.search";
    OpenSearchPartitionInstanceType["m4_xlarge_search"] = "m4.xlarge.search";
    OpenSearchPartitionInstanceType["m5_12xlarge_search"] = "m5.12xlarge.search";
    OpenSearchPartitionInstanceType["m5_24xlarge_search"] = "m5.24xlarge.search";
    OpenSearchPartitionInstanceType["m5_2xlarge_search"] = "m5.2xlarge.search";
    OpenSearchPartitionInstanceType["m5_4xlarge_search"] = "m5.4xlarge.search";
    OpenSearchPartitionInstanceType["m5_large_search"] = "m5.large.search";
    OpenSearchPartitionInstanceType["m5_xlarge_search"] = "m5.xlarge.search";
    OpenSearchPartitionInstanceType["m6g_12xlarge_search"] = "m6g.12xlarge.search";
    OpenSearchPartitionInstanceType["m6g_2xlarge_search"] = "m6g.2xlarge.search";
    OpenSearchPartitionInstanceType["m6g_4xlarge_search"] = "m6g.4xlarge.search";
    OpenSearchPartitionInstanceType["m6g_8xlarge_search"] = "m6g.8xlarge.search";
    OpenSearchPartitionInstanceType["m6g_large_search"] = "m6g.large.search";
    OpenSearchPartitionInstanceType["m6g_xlarge_search"] = "m6g.xlarge.search";
    OpenSearchPartitionInstanceType["r3_2xlarge_search"] = "r3.2xlarge.search";
    OpenSearchPartitionInstanceType["r3_4xlarge_search"] = "r3.4xlarge.search";
    OpenSearchPartitionInstanceType["r3_8xlarge_search"] = "r3.8xlarge.search";
    OpenSearchPartitionInstanceType["r3_large_search"] = "r3.large.search";
    OpenSearchPartitionInstanceType["r3_xlarge_search"] = "r3.xlarge.search";
    OpenSearchPartitionInstanceType["r4_16xlarge_search"] = "r4.16xlarge.search";
    OpenSearchPartitionInstanceType["r4_2xlarge_search"] = "r4.2xlarge.search";
    OpenSearchPartitionInstanceType["r4_4xlarge_search"] = "r4.4xlarge.search";
    OpenSearchPartitionInstanceType["r4_8xlarge_search"] = "r4.8xlarge.search";
    OpenSearchPartitionInstanceType["r4_large_search"] = "r4.large.search";
    OpenSearchPartitionInstanceType["r4_xlarge_search"] = "r4.xlarge.search";
    OpenSearchPartitionInstanceType["r5_12xlarge_search"] = "r5.12xlarge.search";
    OpenSearchPartitionInstanceType["r5_24xlarge_search"] = "r5.24xlarge.search";
    OpenSearchPartitionInstanceType["r5_2xlarge_search"] = "r5.2xlarge.search";
    OpenSearchPartitionInstanceType["r5_4xlarge_search"] = "r5.4xlarge.search";
    OpenSearchPartitionInstanceType["r5_large_search"] = "r5.large.search";
    OpenSearchPartitionInstanceType["r5_xlarge_search"] = "r5.xlarge.search";
    OpenSearchPartitionInstanceType["r6g_12xlarge_search"] = "r6g.12xlarge.search";
    OpenSearchPartitionInstanceType["r6g_2xlarge_search"] = "r6g.2xlarge.search";
    OpenSearchPartitionInstanceType["r6g_4xlarge_search"] = "r6g.4xlarge.search";
    OpenSearchPartitionInstanceType["r6g_8xlarge_search"] = "r6g.8xlarge.search";
    OpenSearchPartitionInstanceType["r6g_large_search"] = "r6g.large.search";
    OpenSearchPartitionInstanceType["r6g_xlarge_search"] = "r6g.xlarge.search";
    OpenSearchPartitionInstanceType["r6gd_12xlarge_search"] = "r6gd.12xlarge.search";
    OpenSearchPartitionInstanceType["r6gd_16xlarge_search"] = "r6gd.16xlarge.search";
    OpenSearchPartitionInstanceType["r6gd_2xlarge_search"] = "r6gd.2xlarge.search";
    OpenSearchPartitionInstanceType["r6gd_4xlarge_search"] = "r6gd.4xlarge.search";
    OpenSearchPartitionInstanceType["r6gd_8xlarge_search"] = "r6gd.8xlarge.search";
    OpenSearchPartitionInstanceType["r6gd_large_search"] = "r6gd.large.search";
    OpenSearchPartitionInstanceType["r6gd_xlarge_search"] = "r6gd.xlarge.search";
    OpenSearchPartitionInstanceType["t2_medium_search"] = "t2.medium.search";
    OpenSearchPartitionInstanceType["t2_micro_search"] = "t2.micro.search";
    OpenSearchPartitionInstanceType["t2_small_search"] = "t2.small.search";
    OpenSearchPartitionInstanceType["t3_2xlarge_search"] = "t3.2xlarge.search";
    OpenSearchPartitionInstanceType["t3_large_search"] = "t3.large.search";
    OpenSearchPartitionInstanceType["t3_medium_search"] = "t3.medium.search";
    OpenSearchPartitionInstanceType["t3_micro_search"] = "t3.micro.search";
    OpenSearchPartitionInstanceType["t3_nano_search"] = "t3.nano.search";
    OpenSearchPartitionInstanceType["t3_small_search"] = "t3.small.search";
    OpenSearchPartitionInstanceType["t3_xlarge_search"] = "t3.xlarge.search";
    OpenSearchPartitionInstanceType["t4g_medium_search"] = "t4g.medium.search";
    OpenSearchPartitionInstanceType["t4g_small_search"] = "t4g.small.search";
    OpenSearchPartitionInstanceType["ultrawarm1_large_search"] = "ultrawarm1.large.search";
    OpenSearchPartitionInstanceType["ultrawarm1_medium_search"] = "ultrawarm1.medium.search";
    OpenSearchPartitionInstanceType["ultrawarm1_xlarge_search"] = "ultrawarm1.xlarge.search";
})(OpenSearchPartitionInstanceType || (OpenSearchPartitionInstanceType = {}));
export var OpenSearchWarmPartitionInstanceType;
(function (OpenSearchWarmPartitionInstanceType) {
    OpenSearchWarmPartitionInstanceType["ultrawarm1_large_search"] = "ultrawarm1.large.search";
    OpenSearchWarmPartitionInstanceType["ultrawarm1_medium_search"] = "ultrawarm1.medium.search";
    OpenSearchWarmPartitionInstanceType["ultrawarm1_xlarge_search"] = "ultrawarm1.xlarge.search";
})(OpenSearchWarmPartitionInstanceType || (OpenSearchWarmPartitionInstanceType = {}));
export var ZoneAwarenessConfig;
(function (ZoneAwarenessConfig) {
    ZoneAwarenessConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ZoneAwarenessConfig || (ZoneAwarenessConfig = {}));
export var ClusterConfig;
(function (ClusterConfig) {
    ClusterConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ClusterConfig || (ClusterConfig = {}));
export var CognitoOptions;
(function (CognitoOptions) {
    CognitoOptions.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CognitoOptions || (CognitoOptions = {}));
export var TLSSecurityPolicy;
(function (TLSSecurityPolicy) {
    TLSSecurityPolicy["POLICY_MIN_TLS_1_0_2019_07"] = "Policy-Min-TLS-1-0-2019-07";
    TLSSecurityPolicy["POLICY_MIN_TLS_1_2_2019_07"] = "Policy-Min-TLS-1-2-2019-07";
})(TLSSecurityPolicy || (TLSSecurityPolicy = {}));
export var DomainEndpointOptions;
(function (DomainEndpointOptions) {
    DomainEndpointOptions.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DomainEndpointOptions || (DomainEndpointOptions = {}));
export var VolumeType;
(function (VolumeType) {
    VolumeType["gp2"] = "gp2";
    VolumeType["io1"] = "io1";
    VolumeType["standard"] = "standard";
})(VolumeType || (VolumeType = {}));
export var EBSOptions;
(function (EBSOptions) {
    EBSOptions.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EBSOptions || (EBSOptions = {}));
export var EncryptionAtRestOptions;
(function (EncryptionAtRestOptions) {
    EncryptionAtRestOptions.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EncryptionAtRestOptions || (EncryptionAtRestOptions = {}));
export var LogType;
(function (LogType) {
    LogType["AUDIT_LOGS"] = "AUDIT_LOGS";
    LogType["ES_APPLICATION_LOGS"] = "ES_APPLICATION_LOGS";
    LogType["INDEX_SLOW_LOGS"] = "INDEX_SLOW_LOGS";
    LogType["SEARCH_SLOW_LOGS"] = "SEARCH_SLOW_LOGS";
})(LogType || (LogType = {}));
export var LogPublishingOption;
(function (LogPublishingOption) {
    LogPublishingOption.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LogPublishingOption || (LogPublishingOption = {}));
export var NodeToNodeEncryptionOptions;
(function (NodeToNodeEncryptionOptions) {
    NodeToNodeEncryptionOptions.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(NodeToNodeEncryptionOptions || (NodeToNodeEncryptionOptions = {}));
export var SnapshotOptions;
(function (SnapshotOptions) {
    SnapshotOptions.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SnapshotOptions || (SnapshotOptions = {}));
export var VPCOptions;
(function (VPCOptions) {
    VPCOptions.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(VPCOptions || (VPCOptions = {}));
export var CreateDomainRequest;
(function (CreateDomainRequest) {
    CreateDomainRequest.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.AdvancedSecurityOptions && {
        AdvancedSecurityOptions: AdvancedSecurityOptionsInput.filterSensitiveLog(obj.AdvancedSecurityOptions),
    }))); };
})(CreateDomainRequest || (CreateDomainRequest = {}));
export var AutoTuneState;
(function (AutoTuneState) {
    AutoTuneState["DISABLED"] = "DISABLED";
    AutoTuneState["DISABLED_AND_ROLLBACK_COMPLETE"] = "DISABLED_AND_ROLLBACK_COMPLETE";
    AutoTuneState["DISABLED_AND_ROLLBACK_ERROR"] = "DISABLED_AND_ROLLBACK_ERROR";
    AutoTuneState["DISABLED_AND_ROLLBACK_IN_PROGRESS"] = "DISABLED_AND_ROLLBACK_IN_PROGRESS";
    AutoTuneState["DISABLED_AND_ROLLBACK_SCHEDULED"] = "DISABLED_AND_ROLLBACK_SCHEDULED";
    AutoTuneState["DISABLE_IN_PROGRESS"] = "DISABLE_IN_PROGRESS";
    AutoTuneState["ENABLED"] = "ENABLED";
    AutoTuneState["ENABLE_IN_PROGRESS"] = "ENABLE_IN_PROGRESS";
    AutoTuneState["ERROR"] = "ERROR";
})(AutoTuneState || (AutoTuneState = {}));
export var AutoTuneOptionsOutput;
(function (AutoTuneOptionsOutput) {
    AutoTuneOptionsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AutoTuneOptionsOutput || (AutoTuneOptionsOutput = {}));
export var VPCDerivedInfo;
(function (VPCDerivedInfo) {
    VPCDerivedInfo.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(VPCDerivedInfo || (VPCDerivedInfo = {}));
export var DomainStatus;
(function (DomainStatus) {
    DomainStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DomainStatus || (DomainStatus = {}));
export var CreateDomainResponse;
(function (CreateDomainResponse) {
    CreateDomainResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateDomainResponse || (CreateDomainResponse = {}));
export var InvalidTypeException;
(function (InvalidTypeException) {
    InvalidTypeException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidTypeException || (InvalidTypeException = {}));
export var ResourceAlreadyExistsException;
(function (ResourceAlreadyExistsException) {
    ResourceAlreadyExistsException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ResourceAlreadyExistsException || (ResourceAlreadyExistsException = {}));
export var CreateOutboundConnectionRequest;
(function (CreateOutboundConnectionRequest) {
    CreateOutboundConnectionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateOutboundConnectionRequest || (CreateOutboundConnectionRequest = {}));
export var OutboundConnectionStatusCode;
(function (OutboundConnectionStatusCode) {
    OutboundConnectionStatusCode["ACTIVE"] = "ACTIVE";
    OutboundConnectionStatusCode["APPROVED"] = "APPROVED";
    OutboundConnectionStatusCode["DELETED"] = "DELETED";
    OutboundConnectionStatusCode["DELETING"] = "DELETING";
    OutboundConnectionStatusCode["PENDING_ACCEPTANCE"] = "PENDING_ACCEPTANCE";
    OutboundConnectionStatusCode["PROVISIONING"] = "PROVISIONING";
    OutboundConnectionStatusCode["REJECTED"] = "REJECTED";
    OutboundConnectionStatusCode["REJECTING"] = "REJECTING";
    OutboundConnectionStatusCode["VALIDATING"] = "VALIDATING";
    OutboundConnectionStatusCode["VALIDATION_FAILED"] = "VALIDATION_FAILED";
})(OutboundConnectionStatusCode || (OutboundConnectionStatusCode = {}));
export var OutboundConnectionStatus;
(function (OutboundConnectionStatus) {
    OutboundConnectionStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(OutboundConnectionStatus || (OutboundConnectionStatus = {}));
export var CreateOutboundConnectionResponse;
(function (CreateOutboundConnectionResponse) {
    CreateOutboundConnectionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateOutboundConnectionResponse || (CreateOutboundConnectionResponse = {}));
export var PackageSource;
(function (PackageSource) {
    PackageSource.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PackageSource || (PackageSource = {}));
export var CreatePackageRequest;
(function (CreatePackageRequest) {
    CreatePackageRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreatePackageRequest || (CreatePackageRequest = {}));
export var PackageStatus;
(function (PackageStatus) {
    PackageStatus["AVAILABLE"] = "AVAILABLE";
    PackageStatus["COPYING"] = "COPYING";
    PackageStatus["COPY_FAILED"] = "COPY_FAILED";
    PackageStatus["DELETED"] = "DELETED";
    PackageStatus["DELETE_FAILED"] = "DELETE_FAILED";
    PackageStatus["DELETING"] = "DELETING";
    PackageStatus["VALIDATING"] = "VALIDATING";
    PackageStatus["VALIDATION_FAILED"] = "VALIDATION_FAILED";
})(PackageStatus || (PackageStatus = {}));
export var PackageDetails;
(function (PackageDetails) {
    PackageDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PackageDetails || (PackageDetails = {}));
export var CreatePackageResponse;
(function (CreatePackageResponse) {
    CreatePackageResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreatePackageResponse || (CreatePackageResponse = {}));
export var DeleteDomainRequest;
(function (DeleteDomainRequest) {
    DeleteDomainRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDomainRequest || (DeleteDomainRequest = {}));
export var DeleteDomainResponse;
(function (DeleteDomainResponse) {
    DeleteDomainResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteDomainResponse || (DeleteDomainResponse = {}));
export var DeleteInboundConnectionRequest;
(function (DeleteInboundConnectionRequest) {
    DeleteInboundConnectionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteInboundConnectionRequest || (DeleteInboundConnectionRequest = {}));
export var DeleteInboundConnectionResponse;
(function (DeleteInboundConnectionResponse) {
    DeleteInboundConnectionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteInboundConnectionResponse || (DeleteInboundConnectionResponse = {}));
export var DeleteOutboundConnectionRequest;
(function (DeleteOutboundConnectionRequest) {
    DeleteOutboundConnectionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteOutboundConnectionRequest || (DeleteOutboundConnectionRequest = {}));
export var OutboundConnection;
(function (OutboundConnection) {
    OutboundConnection.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(OutboundConnection || (OutboundConnection = {}));
export var DeleteOutboundConnectionResponse;
(function (DeleteOutboundConnectionResponse) {
    DeleteOutboundConnectionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteOutboundConnectionResponse || (DeleteOutboundConnectionResponse = {}));
export var DeletePackageRequest;
(function (DeletePackageRequest) {
    DeletePackageRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeletePackageRequest || (DeletePackageRequest = {}));
export var DeletePackageResponse;
(function (DeletePackageResponse) {
    DeletePackageResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeletePackageResponse || (DeletePackageResponse = {}));
export var DescribeDomainRequest;
(function (DescribeDomainRequest) {
    DescribeDomainRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDomainRequest || (DescribeDomainRequest = {}));
export var DescribeDomainResponse;
(function (DescribeDomainResponse) {
    DescribeDomainResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDomainResponse || (DescribeDomainResponse = {}));
export var DescribeDomainAutoTunesRequest;
(function (DescribeDomainAutoTunesRequest) {
    DescribeDomainAutoTunesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDomainAutoTunesRequest || (DescribeDomainAutoTunesRequest = {}));
export var ScheduledAutoTuneActionType;
(function (ScheduledAutoTuneActionType) {
    ScheduledAutoTuneActionType["JVM_HEAP_SIZE_TUNING"] = "JVM_HEAP_SIZE_TUNING";
    ScheduledAutoTuneActionType["JVM_YOUNG_GEN_TUNING"] = "JVM_YOUNG_GEN_TUNING";
})(ScheduledAutoTuneActionType || (ScheduledAutoTuneActionType = {}));
export var ScheduledAutoTuneSeverityType;
(function (ScheduledAutoTuneSeverityType) {
    ScheduledAutoTuneSeverityType["HIGH"] = "HIGH";
    ScheduledAutoTuneSeverityType["LOW"] = "LOW";
    ScheduledAutoTuneSeverityType["MEDIUM"] = "MEDIUM";
})(ScheduledAutoTuneSeverityType || (ScheduledAutoTuneSeverityType = {}));
export var ScheduledAutoTuneDetails;
(function (ScheduledAutoTuneDetails) {
    ScheduledAutoTuneDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ScheduledAutoTuneDetails || (ScheduledAutoTuneDetails = {}));
export var AutoTuneDetails;
(function (AutoTuneDetails) {
    AutoTuneDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AutoTuneDetails || (AutoTuneDetails = {}));
export var AutoTuneType;
(function (AutoTuneType) {
    AutoTuneType["SCHEDULED_ACTION"] = "SCHEDULED_ACTION";
})(AutoTuneType || (AutoTuneType = {}));
export var AutoTune;
(function (AutoTune) {
    AutoTune.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AutoTune || (AutoTune = {}));
export var DescribeDomainAutoTunesResponse;
(function (DescribeDomainAutoTunesResponse) {
    DescribeDomainAutoTunesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDomainAutoTunesResponse || (DescribeDomainAutoTunesResponse = {}));
export var DescribeDomainConfigRequest;
(function (DescribeDomainConfigRequest) {
    DescribeDomainConfigRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDomainConfigRequest || (DescribeDomainConfigRequest = {}));
export var RollbackOnDisable;
(function (RollbackOnDisable) {
    RollbackOnDisable["DEFAULT_ROLLBACK"] = "DEFAULT_ROLLBACK";
    RollbackOnDisable["NO_ROLLBACK"] = "NO_ROLLBACK";
})(RollbackOnDisable || (RollbackOnDisable = {}));
export var AutoTuneOptions;
(function (AutoTuneOptions) {
    AutoTuneOptions.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AutoTuneOptions || (AutoTuneOptions = {}));
export var AutoTuneStatus;
(function (AutoTuneStatus) {
    AutoTuneStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AutoTuneStatus || (AutoTuneStatus = {}));
export var AutoTuneOptionsStatus;
(function (AutoTuneOptionsStatus) {
    AutoTuneOptionsStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AutoTuneOptionsStatus || (AutoTuneOptionsStatus = {}));
export var ClusterConfigStatus;
(function (ClusterConfigStatus) {
    ClusterConfigStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ClusterConfigStatus || (ClusterConfigStatus = {}));
export var CognitoOptionsStatus;
(function (CognitoOptionsStatus) {
    CognitoOptionsStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CognitoOptionsStatus || (CognitoOptionsStatus = {}));
export var DomainEndpointOptionsStatus;
(function (DomainEndpointOptionsStatus) {
    DomainEndpointOptionsStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DomainEndpointOptionsStatus || (DomainEndpointOptionsStatus = {}));
export var EBSOptionsStatus;
(function (EBSOptionsStatus) {
    EBSOptionsStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EBSOptionsStatus || (EBSOptionsStatus = {}));
export var EncryptionAtRestOptionsStatus;
(function (EncryptionAtRestOptionsStatus) {
    EncryptionAtRestOptionsStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(EncryptionAtRestOptionsStatus || (EncryptionAtRestOptionsStatus = {}));
export var VersionStatus;
(function (VersionStatus) {
    VersionStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(VersionStatus || (VersionStatus = {}));
export var LogPublishingOptionsStatus;
(function (LogPublishingOptionsStatus) {
    LogPublishingOptionsStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LogPublishingOptionsStatus || (LogPublishingOptionsStatus = {}));
export var NodeToNodeEncryptionOptionsStatus;
(function (NodeToNodeEncryptionOptionsStatus) {
    NodeToNodeEncryptionOptionsStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(NodeToNodeEncryptionOptionsStatus || (NodeToNodeEncryptionOptionsStatus = {}));
export var SnapshotOptionsStatus;
(function (SnapshotOptionsStatus) {
    SnapshotOptionsStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SnapshotOptionsStatus || (SnapshotOptionsStatus = {}));
export var VPCDerivedInfoStatus;
(function (VPCDerivedInfoStatus) {
    VPCDerivedInfoStatus.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(VPCDerivedInfoStatus || (VPCDerivedInfoStatus = {}));
export var DomainConfig;
(function (DomainConfig) {
    DomainConfig.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DomainConfig || (DomainConfig = {}));
export var DescribeDomainConfigResponse;
(function (DescribeDomainConfigResponse) {
    DescribeDomainConfigResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDomainConfigResponse || (DescribeDomainConfigResponse = {}));
export var DescribeDomainsRequest;
(function (DescribeDomainsRequest) {
    DescribeDomainsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDomainsRequest || (DescribeDomainsRequest = {}));
export var DescribeDomainsResponse;
(function (DescribeDomainsResponse) {
    DescribeDomainsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeDomainsResponse || (DescribeDomainsResponse = {}));
export var Filter;
(function (Filter) {
    Filter.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Filter || (Filter = {}));
export var DescribeInboundConnectionsRequest;
(function (DescribeInboundConnectionsRequest) {
    DescribeInboundConnectionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeInboundConnectionsRequest || (DescribeInboundConnectionsRequest = {}));
export var DescribeInboundConnectionsResponse;
(function (DescribeInboundConnectionsResponse) {
    DescribeInboundConnectionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeInboundConnectionsResponse || (DescribeInboundConnectionsResponse = {}));
export var InvalidPaginationTokenException;
(function (InvalidPaginationTokenException) {
    InvalidPaginationTokenException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidPaginationTokenException || (InvalidPaginationTokenException = {}));
export var DescribeInstanceTypeLimitsRequest;
(function (DescribeInstanceTypeLimitsRequest) {
    DescribeInstanceTypeLimitsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeInstanceTypeLimitsRequest || (DescribeInstanceTypeLimitsRequest = {}));
export var InstanceCountLimits;
(function (InstanceCountLimits) {
    InstanceCountLimits.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InstanceCountLimits || (InstanceCountLimits = {}));
export var InstanceLimits;
(function (InstanceLimits) {
    InstanceLimits.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InstanceLimits || (InstanceLimits = {}));
export var StorageTypeLimit;
(function (StorageTypeLimit) {
    StorageTypeLimit.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StorageTypeLimit || (StorageTypeLimit = {}));
export var StorageType;
(function (StorageType) {
    StorageType.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StorageType || (StorageType = {}));
export var Limits;
(function (Limits) {
    Limits.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Limits || (Limits = {}));
export var DescribeInstanceTypeLimitsResponse;
(function (DescribeInstanceTypeLimitsResponse) {
    DescribeInstanceTypeLimitsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeInstanceTypeLimitsResponse || (DescribeInstanceTypeLimitsResponse = {}));
export var DescribeOutboundConnectionsRequest;
(function (DescribeOutboundConnectionsRequest) {
    DescribeOutboundConnectionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeOutboundConnectionsRequest || (DescribeOutboundConnectionsRequest = {}));
export var DescribeOutboundConnectionsResponse;
(function (DescribeOutboundConnectionsResponse) {
    DescribeOutboundConnectionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeOutboundConnectionsResponse || (DescribeOutboundConnectionsResponse = {}));
export var DescribePackagesFilterName;
(function (DescribePackagesFilterName) {
    DescribePackagesFilterName["PackageID"] = "PackageID";
    DescribePackagesFilterName["PackageName"] = "PackageName";
    DescribePackagesFilterName["PackageStatus"] = "PackageStatus";
})(DescribePackagesFilterName || (DescribePackagesFilterName = {}));
export var DescribePackagesFilter;
(function (DescribePackagesFilter) {
    DescribePackagesFilter.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribePackagesFilter || (DescribePackagesFilter = {}));
export var DescribePackagesRequest;
(function (DescribePackagesRequest) {
    DescribePackagesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribePackagesRequest || (DescribePackagesRequest = {}));
export var DescribePackagesResponse;
(function (DescribePackagesResponse) {
    DescribePackagesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribePackagesResponse || (DescribePackagesResponse = {}));
export var DescribeReservedInstanceOfferingsRequest;
(function (DescribeReservedInstanceOfferingsRequest) {
    DescribeReservedInstanceOfferingsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeReservedInstanceOfferingsRequest || (DescribeReservedInstanceOfferingsRequest = {}));
export var ReservedInstancePaymentOption;
(function (ReservedInstancePaymentOption) {
    ReservedInstancePaymentOption["ALL_UPFRONT"] = "ALL_UPFRONT";
    ReservedInstancePaymentOption["NO_UPFRONT"] = "NO_UPFRONT";
    ReservedInstancePaymentOption["PARTIAL_UPFRONT"] = "PARTIAL_UPFRONT";
})(ReservedInstancePaymentOption || (ReservedInstancePaymentOption = {}));
export var RecurringCharge;
(function (RecurringCharge) {
    RecurringCharge.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RecurringCharge || (RecurringCharge = {}));
export var ReservedInstanceOffering;
(function (ReservedInstanceOffering) {
    ReservedInstanceOffering.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReservedInstanceOffering || (ReservedInstanceOffering = {}));
export var DescribeReservedInstanceOfferingsResponse;
(function (DescribeReservedInstanceOfferingsResponse) {
    DescribeReservedInstanceOfferingsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeReservedInstanceOfferingsResponse || (DescribeReservedInstanceOfferingsResponse = {}));
export var DescribeReservedInstancesRequest;
(function (DescribeReservedInstancesRequest) {
    DescribeReservedInstancesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeReservedInstancesRequest || (DescribeReservedInstancesRequest = {}));
export var ReservedInstance;
(function (ReservedInstance) {
    ReservedInstance.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReservedInstance || (ReservedInstance = {}));
export var DescribeReservedInstancesResponse;
(function (DescribeReservedInstancesResponse) {
    DescribeReservedInstancesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeReservedInstancesResponse || (DescribeReservedInstancesResponse = {}));
export var DissociatePackageRequest;
(function (DissociatePackageRequest) {
    DissociatePackageRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DissociatePackageRequest || (DissociatePackageRequest = {}));
export var DissociatePackageResponse;
(function (DissociatePackageResponse) {
    DissociatePackageResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DissociatePackageResponse || (DissociatePackageResponse = {}));
export var GetCompatibleVersionsRequest;
(function (GetCompatibleVersionsRequest) {
    GetCompatibleVersionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetCompatibleVersionsRequest || (GetCompatibleVersionsRequest = {}));
export var CompatibleVersionsMap;
(function (CompatibleVersionsMap) {
    CompatibleVersionsMap.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CompatibleVersionsMap || (CompatibleVersionsMap = {}));
export var GetCompatibleVersionsResponse;
(function (GetCompatibleVersionsResponse) {
    GetCompatibleVersionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetCompatibleVersionsResponse || (GetCompatibleVersionsResponse = {}));
export var GetPackageVersionHistoryRequest;
(function (GetPackageVersionHistoryRequest) {
    GetPackageVersionHistoryRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetPackageVersionHistoryRequest || (GetPackageVersionHistoryRequest = {}));
export var PackageVersionHistory;
(function (PackageVersionHistory) {
    PackageVersionHistory.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PackageVersionHistory || (PackageVersionHistory = {}));
export var GetPackageVersionHistoryResponse;
(function (GetPackageVersionHistoryResponse) {
    GetPackageVersionHistoryResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetPackageVersionHistoryResponse || (GetPackageVersionHistoryResponse = {}));
export var GetUpgradeHistoryRequest;
(function (GetUpgradeHistoryRequest) {
    GetUpgradeHistoryRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetUpgradeHistoryRequest || (GetUpgradeHistoryRequest = {}));
export var UpgradeStep;
(function (UpgradeStep) {
    UpgradeStep["PRE_UPGRADE_CHECK"] = "PRE_UPGRADE_CHECK";
    UpgradeStep["SNAPSHOT"] = "SNAPSHOT";
    UpgradeStep["UPGRADE"] = "UPGRADE";
})(UpgradeStep || (UpgradeStep = {}));
export var UpgradeStatus;
(function (UpgradeStatus) {
    UpgradeStatus["FAILED"] = "FAILED";
    UpgradeStatus["IN_PROGRESS"] = "IN_PROGRESS";
    UpgradeStatus["SUCCEEDED"] = "SUCCEEDED";
    UpgradeStatus["SUCCEEDED_WITH_ISSUES"] = "SUCCEEDED_WITH_ISSUES";
})(UpgradeStatus || (UpgradeStatus = {}));
export var UpgradeStepItem;
(function (UpgradeStepItem) {
    UpgradeStepItem.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpgradeStepItem || (UpgradeStepItem = {}));
export var UpgradeHistory;
(function (UpgradeHistory) {
    UpgradeHistory.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpgradeHistory || (UpgradeHistory = {}));
export var GetUpgradeHistoryResponse;
(function (GetUpgradeHistoryResponse) {
    GetUpgradeHistoryResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetUpgradeHistoryResponse || (GetUpgradeHistoryResponse = {}));
export var GetUpgradeStatusRequest;
(function (GetUpgradeStatusRequest) {
    GetUpgradeStatusRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetUpgradeStatusRequest || (GetUpgradeStatusRequest = {}));
export var GetUpgradeStatusResponse;
(function (GetUpgradeStatusResponse) {
    GetUpgradeStatusResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GetUpgradeStatusResponse || (GetUpgradeStatusResponse = {}));
export var EngineType;
(function (EngineType) {
    EngineType["Elasticsearch"] = "Elasticsearch";
    EngineType["OpenSearch"] = "OpenSearch";
})(EngineType || (EngineType = {}));
export var ListDomainNamesRequest;
(function (ListDomainNamesRequest) {
    ListDomainNamesRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListDomainNamesRequest || (ListDomainNamesRequest = {}));
export var DomainInfo;
(function (DomainInfo) {
    DomainInfo.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DomainInfo || (DomainInfo = {}));
export var ListDomainNamesResponse;
(function (ListDomainNamesResponse) {
    ListDomainNamesResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListDomainNamesResponse || (ListDomainNamesResponse = {}));
export var ListDomainsForPackageRequest;
(function (ListDomainsForPackageRequest) {
    ListDomainsForPackageRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListDomainsForPackageRequest || (ListDomainsForPackageRequest = {}));
export var ListDomainsForPackageResponse;
(function (ListDomainsForPackageResponse) {
    ListDomainsForPackageResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListDomainsForPackageResponse || (ListDomainsForPackageResponse = {}));
export var ListInstanceTypeDetailsRequest;
(function (ListInstanceTypeDetailsRequest) {
    ListInstanceTypeDetailsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListInstanceTypeDetailsRequest || (ListInstanceTypeDetailsRequest = {}));
export var InstanceTypeDetails;
(function (InstanceTypeDetails) {
    InstanceTypeDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InstanceTypeDetails || (InstanceTypeDetails = {}));
export var ListInstanceTypeDetailsResponse;
(function (ListInstanceTypeDetailsResponse) {
    ListInstanceTypeDetailsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListInstanceTypeDetailsResponse || (ListInstanceTypeDetailsResponse = {}));
export var ListPackagesForDomainRequest;
(function (ListPackagesForDomainRequest) {
    ListPackagesForDomainRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListPackagesForDomainRequest || (ListPackagesForDomainRequest = {}));
export var ListPackagesForDomainResponse;
(function (ListPackagesForDomainResponse) {
    ListPackagesForDomainResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListPackagesForDomainResponse || (ListPackagesForDomainResponse = {}));
export var ListTagsRequest;
(function (ListTagsRequest) {
    ListTagsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTagsRequest || (ListTagsRequest = {}));
export var ListTagsResponse;
(function (ListTagsResponse) {
    ListTagsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTagsResponse || (ListTagsResponse = {}));
export var ListVersionsRequest;
(function (ListVersionsRequest) {
    ListVersionsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListVersionsRequest || (ListVersionsRequest = {}));
export var ListVersionsResponse;
(function (ListVersionsResponse) {
    ListVersionsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListVersionsResponse || (ListVersionsResponse = {}));
export var PurchaseReservedInstanceOfferingRequest;
(function (PurchaseReservedInstanceOfferingRequest) {
    PurchaseReservedInstanceOfferingRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PurchaseReservedInstanceOfferingRequest || (PurchaseReservedInstanceOfferingRequest = {}));
export var PurchaseReservedInstanceOfferingResponse;
(function (PurchaseReservedInstanceOfferingResponse) {
    PurchaseReservedInstanceOfferingResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PurchaseReservedInstanceOfferingResponse || (PurchaseReservedInstanceOfferingResponse = {}));
export var RejectInboundConnectionRequest;
(function (RejectInboundConnectionRequest) {
    RejectInboundConnectionRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RejectInboundConnectionRequest || (RejectInboundConnectionRequest = {}));
export var RejectInboundConnectionResponse;
(function (RejectInboundConnectionResponse) {
    RejectInboundConnectionResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RejectInboundConnectionResponse || (RejectInboundConnectionResponse = {}));
export var RemoveTagsRequest;
(function (RemoveTagsRequest) {
    RemoveTagsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RemoveTagsRequest || (RemoveTagsRequest = {}));
export var StartServiceSoftwareUpdateRequest;
(function (StartServiceSoftwareUpdateRequest) {
    StartServiceSoftwareUpdateRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StartServiceSoftwareUpdateRequest || (StartServiceSoftwareUpdateRequest = {}));
export var StartServiceSoftwareUpdateResponse;
(function (StartServiceSoftwareUpdateResponse) {
    StartServiceSoftwareUpdateResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StartServiceSoftwareUpdateResponse || (StartServiceSoftwareUpdateResponse = {}));
export var UpdateDomainConfigRequest;
(function (UpdateDomainConfigRequest) {
    UpdateDomainConfigRequest.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.AdvancedSecurityOptions && {
        AdvancedSecurityOptions: AdvancedSecurityOptionsInput.filterSensitiveLog(obj.AdvancedSecurityOptions),
    }))); };
})(UpdateDomainConfigRequest || (UpdateDomainConfigRequest = {}));
export var UpdateDomainConfigResponse;
(function (UpdateDomainConfigResponse) {
    UpdateDomainConfigResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateDomainConfigResponse || (UpdateDomainConfigResponse = {}));
export var UpdatePackageRequest;
(function (UpdatePackageRequest) {
    UpdatePackageRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdatePackageRequest || (UpdatePackageRequest = {}));
export var UpdatePackageResponse;
(function (UpdatePackageResponse) {
    UpdatePackageResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdatePackageResponse || (UpdatePackageResponse = {}));
export var UpgradeDomainRequest;
(function (UpgradeDomainRequest) {
    UpgradeDomainRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpgradeDomainRequest || (UpgradeDomainRequest = {}));
export var UpgradeDomainResponse;
(function (UpgradeDomainResponse) {
    UpgradeDomainResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpgradeDomainResponse || (UpgradeDomainResponse = {}));
