// @aws-sdk/client-opensearch/dist/es/commands/RejectInboundConnectionCommand.js
import { __extends } from "tslib";
import { RejectInboundConnectionRequest, RejectInboundConnectionResponse } from "../models/models_0";
import { deserializeAws_restJson1RejectInboundConnectionCommand, serializeAws_restJson1RejectInboundConnectionCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
var RejectInboundConnectionCommand = (function (_super) {
    __extends(RejectInboundConnectionCommand, _super);
    function RejectInboundConnectionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    RejectInboundConnectionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "OpenSearchClient";
        var commandName = "RejectInboundConnectionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: RejectInboundConnectionRequest.filterSensitiveLog,
            outputFilterSensitiveLog: RejectInboundConnectionResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    RejectInboundConnectionCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1RejectInboundConnectionCommand(input, context);
    };
    RejectInboundConnectionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1RejectInboundConnectionCommand(output, context);
    };
    return RejectInboundConnectionCommand;
}($Command));
export { RejectInboundConnectionCommand };
