// @aws-sdk/client-opensearch/dist/es/commands/DescribeOutboundConnectionsCommand.js
import { __extends } from "tslib";
import { DescribeOutboundConnectionsRequest, DescribeOutboundConnectionsResponse } from "../models/models_0";
import { deserializeAws_restJson1DescribeOutboundConnectionsCommand, serializeAws_restJson1DescribeOutboundConnectionsCommand, } from "../protocols/Aws_restJson1";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
var DescribeOutboundConnectionsCommand = (function (_super) {
    __extends(DescribeOutboundConnectionsCommand, _super);
    function DescribeOutboundConnectionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DescribeOutboundConnectionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "OpenSearchClient";
        var commandName = "DescribeOutboundConnectionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DescribeOutboundConnectionsRequest.filterSensitiveLog,
            outputFilterSensitiveLog: DescribeOutboundConnectionsResponse.filterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DescribeOutboundConnectionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_restJson1DescribeOutboundConnectionsCommand(input, context);
    };
    DescribeOutboundConnectionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_restJson1DescribeOutboundConnectionsCommand(output, context);
    };
    return DescribeOutboundConnectionsCommand;
}($Command));
export { DescribeOutboundConnectionsCommand };
